Lib/test/test_type_aliases.py

```diff
@@ -0,0 +1,137 @@
+import textwrap
+import types
+import unittest
+
+from typing import TypeAliasType
+
+class TypeParamsInvalidTest(unittest.TestCase):
+    def test_name_collision_01(self):
+        code = """type TA1[A, **A] = None"""
+
+        with self.assertRaisesRegex(SyntaxError, "duplicate type parameter 'A'"):
+            exec(code, {}, {})
+
+    def test_name_non_collision_02(self):
+        code = """type TA1[A] = lambda A: None"""
+        exec(code, {}, {})
+
+    def test_name_non_collision_03(self):
+        code = textwrap.dedent("""\
+            class Outer[A]:
+                type TA1[A] = None
+            """
+        )
+        exec(code, {}, {})
```

Comment:
As an alternative to using `exec()` here, you could use doctests like in `test_syntax.py`: https://github.com/python/cpython/blob/main/Lib/test/test_syntax.py

That might make the test more readable

Reply:
I'd prefer to leave this as is for now; we can always change it later.

Reply:
I added a helper as @carljm suggested.

Reply:
Thanks, that makes it much more readable!